packages/sync/src/links/OfflineQueueLink.ts

```diff
@@ -153,4 +171,21 @@ export class OfflineQueueLink extends ApolloLink {
       });
     }
   }
+
+  private updateIds(
+    operationEntry: OperationQueueEntry,
+    result: FetchResult
+  ): void {
+    const operation = operationEntry.operation;
+    const optimisticResponse = operationEntry.optimisticResponse;
+    if (optimisticResponse && optimisticResponse[operation.operationName].__optimisticId) {
+      const optimisticId = optimisticResponse[operation.operationName].id;
+      this.opQueue.forEach(({ operation: op }) => {
+        if (op.variables.id === optimisticId && result.data) {
```

Comment:
I think that if we go with this to avoid id clashes between server and client we need to generate ID's using `-` or with some prefix like `client:3gj4hgj34hbgdfgf`

Reply:
@wtrocki That is a very good catch. I think a prefix is a good solution

Reply:
Yes, I think that if we do this, we can also remove the `__optimisticId` flag.